document-store/src/main/java/org/hypertrace/core/documentstore/postgres/PostgresDatastore.java

```diff
@@ -0,0 +1,144 @@
+package org.hypertrace.core.documentstore.postgres;
+
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.CREATED_AT;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.DOCUMENT;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.ID;
+import static org.hypertrace.core.documentstore.postgres.PostgresCollection.UPDATED_AT;
+
+import com.typesafe.config.Config;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.hypertrace.core.documentstore.Collection;
+import org.hypertrace.core.documentstore.Datastore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides {@link Datastore} implementation for Postgres DB.
+ */
+public class PostgresDatastore implements Datastore {
```

Comment:
Since this is an implementation that I believe it isn't supposed to be extended, does making it final makes sense?

Reply:
@jainnidhi703 Can you handle this in your follow up PR?